autoload/sj/elm.vim

```diff
@@ -0,0 +1,247 @@
+" vim: foldmethod=marker
+
+" Split / Join {{{1
+"
+" function! sj#elm#Join() {{{2
+"
+" Attempts to join the multiline tuple, record or list closest
+" to current curspor position.
+function! sj#elm#Join()
+  let [from, to] = sj#elm#ClosestMultilineBraces()
+
+  if from[0] < 0
+    return 0
+  endif
+
+  let original = sj#GetByPosition([0, from[0], from[1]], [0, to[0], to[1]])
+  let joined = sj#elm#JoinOuterBraces(sj#elm#JoinLines(original))
+  call sj#ReplaceByPosition([0, from[0], from[1]], [0, to[0], to[1]], joined)
+
+  return 1
+endfunction
```

Comment:
I would like to name this function something more specific. Right now, there's only one `split` and `join` callback, and you might pick different names for later callbacks, but these are functions with public access, essentially the "interface" of the plugin. Someone could copy the ftplugin settings in order to remove callbacks or add new ones, and changing the names of these functions would be disruptive.

Since this function references "braces" several times in the implementation, maybe we could call it `JoinBraces`, and `SplitBraces` for the below one. I would also consider something like `JoinList` or `JoinSequence` or something. 

In other filetypes, I slice them even finer, for example in javascript: https://github.com/AndrewRadev/splitjoin.vim/blob/20e41455e1155f5989ecac007fc92c9415244822/autoload/sj/js.vim#L92-L98

Initially, these functions didn't even have a "cursor position" parameter, but the fact that they were separate made it easier to add it later. I wouldn't worry too much about splitting them up now, though -- no need if it's not necessary.